### DISCUSSION TITLE
Smart naming help

### DISCUSSION THREAD

**Discussion by xQuiZ95** [aclap-dev/video-downloadhelper#1414] [Help and Q&A]
### Your question

Been a premium user for a long time and earlier versions managed to get the names correctly on my specific website. Unfortunately now days I'm not able to get the right name at all and it will just take the name that's inside of the tab. 

I'm gonna say this straight away, I've done some HTML and CSS coding but it's very limited. Tried following the limited information that is out there but haven't been able to get it to work. Since none of that is working I would like to just get the URL, preferably the end of the URL to be the name of my file. What should I do to make this happen?

I would also like to add how I can download from the same stream multiple times without having to reload the extension? Reloading will stop my other downloads.

### Addon & CoApp version and details

version: 9.1.1.0
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"5cfb9691f4484c64"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "D:\\Media\\Eget\\Korean\\방송\\Recordings",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "show_success_notification": false,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "forget_media_on_tab_close": true,
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 9,
    "successfull_dl": 5,
    "never_show_successfull_dl_message": true,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
           ,
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2096943810653616"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/zmalqp6666"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729353096867
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_4258613823393548"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-01.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-01.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/o111na"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729353593708
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_3268511869705357"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-03.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-03.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/day59day"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729354525868
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2197511091176212"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-05.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-05.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/qwas33"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729356092616
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_3901220863450745"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-06.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-06.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/in0410"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729358860414
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_1071912969912331"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-07.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-07.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/o111na"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1729359202552
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_547731135601893"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-08.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-08.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/hyhy990"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1730116828080
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_8832089014972963"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "팬더티비-www pandalive co kr-09.mp4"
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\Oliwe\\dwhelper\\팬더티비-www pandalive co kr-09.mp4"
                            },
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.pandalive.co.kr/live/play/muse62"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1730116828297
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}

**Comment by Mattmjs** [comment]
Refer to this for smart naming https://github.com/aclap-dev/video-downloadhelper/discussions/838

Just duplicate the tab for that stream however many times needed to download from the same stream.

**Reply by xQuiZ95** [reply]
This is unfortunately not possible since you can only have one stream open at a time. If I open multiple windows it will kick me out of the stream.

**Comment by xQuiZ95** [comment]
Pretty disappointed that I haven't had a single answer that could help me with this. I bought the program because I wanted an easy way to record streams and download videos but unfortunately it's been way easier for me to learn ffmpeg and yt-dlp and set it up in a way that is just easier and more straight forward for my use case. 

Not sure if the creator of this program will ever read this but I'm not going to recommend this program to anyone going forward.

**Reply by xQuiZ95** [reply]
I also want to add that the program just feels broken and incomplete since you have to go back to using the legacy UI for the majority of what you'd like to do which shouldn't be the case. The new ui should imo be an improvement of the previous one, not a downgrade.
